### ISSUE TITLE
Is there any permission that must be enabled to get webRTC working on Android?

### ISSUE THREAD

**Issue by Nachocss** [godotengine/webrtc-native#45]
Hi, I was trying to figure out why my implementation of the plugin didn't work at all, which shouldn't happen since I'm recovering some code of another webRTC project (in that case, focused on Web environment).

The thing is, I realized it actually works connecting Windows to Windows builds, which can send packets among them, so the problem is setting the network peers on Android targets, which doesn't throw any error but leave the network peers' array empty.

So my question is, do I need to set some permission on the export settings to get it working properly?

Thank you in advance.

**Comment by Faless** [comment]
> Hi, I was trying to figure out why my implementation of the plugin didn't work at all

This seems pretty generic. Can you expand on this? Did you get a crash? Did you get an error? Do you have godot-related errors in `adb logcat`?

> do I need to set some permission on the export settings to get it working properly?

Well, definitely `android.permission.INTERNET`, probably `android.permission.ACCESS_NETWORK_STATE`.

**Comment by Nachocss** [comment]
> This seems pretty generic. Can you expand on this? Did you get a crash? Did you get an error? Do you have godot-related errors in `adb logcat`?

All right, I'm checking logcat and it seems to throw an error with the following message:
`**ERROR**: Default GDNative WebRTC implementation not defined.`

However, GDNative plugins are enabled and webrtc.tres and webrtc_debug.tres are pointing to the correct directory.
![image](https://user-images.githubusercontent.com/36671144/126979379-d5317376-636a-476c-9700-b8e3cf4ac908.png)



> Well, definitely `android.permission.INTERNET`, probably `android.permission.ACCESS_NETWORK_STATE`.

`android.permission.INTERNET` was enabled when I opened the issue. After enabling `android.permission.ACCESS_NETWORK_STATE` I got the same result.

**Comment by Faless** [comment]
Are you using the [latest version](https://github.com/godotengine/webrtc-native/releases) of the plugin? What is your device OS and model?

>  `**ERROR**: Default GDNative WebRTC implementation not defined.`

You should be getting some other godot `**ERROR**`s before from godot before this, can you post those too?

**Comment by Nachocss** [comment]
I'm using [0.5](https://github.com/godotengine/webrtc-native/releases/tag/0.5) version on Godot v3.3.2-stable win64.

**Android version:** 10
**API level:** 29
**Device model:** Redmi Note 8


> You should be getting some other godot `**ERROR**`s before this, can you post those too?

You are right, it's throwing 
```
07-26 12:22:22.567 27134 27333 E godot   : **ERROR**: Can't open dynamic library: libwebrtc_native.android.release.arm64v8.so, error: dlopen failed: library "libwebrtc_native.android.release.arm64v8.so" not found.

07-26 12:22:22.567 27134 27333 E godot   :    At: platform/android/os_android.cpp:228:open_dynamic_library() - Condition "!p_library_handle" is true. Returned: ERR_CANT_OPEN
```

Do I need to place the library elsewhere than _res://webrtc/lib/_ or could it be related to read permissions?

**Comment by Faless** [comment]
> Do I need to place the library elsewhere than _res://webrtc/lib/_ 

No, the plugin zip must be extracted in the root folder of the project. As mentioned in the documentation.

> or could it be related to read permissions?

I doubt it given the error.

---
If you export and open the generate APK do you get the library in the right place?

![android](https://user-images.githubusercontent.com/1687918/126985107-ea972962-e4e3-4576-9214-878ba77c1237.png)

**Comment by Nachocss** [comment]
No it doesn't 
![image](https://user-images.githubusercontent.com/36671144/126985704-ea28e0db-04bb-4b29-8980-8f8837eb7de0.png)

**Comment by Nachocss** [comment]
It seems the problem is related with using a custom build (in order to implement Admob). 
By disabling custom builds, it includes the library where it should be
![image](https://user-images.githubusercontent.com/36671144/127029318-27ac9fd2-5298-44cd-adc6-5cd449cf9490.png)

**Comment by Faless** [comment]
> It seems the problem is related to custom build use _(in order to implement Admob)._

Thanks for reporting back.
The issue seems fixed via https://github.com/godotengine/godot/pull/49912, which will be available in future Godot releases (including `3.3.3` and `3.4`s). Closing as fixed.

**Comment by Nachocss** [comment]
For the record, build is working correctly on Godot 3.4.beta1 without further change than Godot version

Thank you @Faless Really appreciated.
